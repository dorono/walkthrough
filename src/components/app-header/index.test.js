import React from 'react';
import {shallow} from 'enzyme';
import AppHeader from './index';

const apiConfig = {
    blockchain: 'Public',
};

const defaultProps = {
    apiConfig,
    notifyRemoteConfigWasBlocked: false,
    defaultApiConfig: apiConfig,
    allowCustomCredentials: false,
    isConfiguredByDefault: jest.fn,
    remoteConfig: false,
    onSettingsSubmit: jest.fn,
};

describe('App Header Component', () => {
    it('should render without errors', () => {
        shallow(<AppHeader {...defaultProps} />);
    });

    it('should display the Settings Popup if notifyRemoteConfigWasBlocked is true', () => {
        const appHeader = shallow(<AppHeader {...defaultProps} notifyRemoteConfigWasBlocked />);
        expect(appHeader.find('SettingsPopup').length).toBe(1);
        expect(appHeader).toMatchSnapshot();
    });
});
