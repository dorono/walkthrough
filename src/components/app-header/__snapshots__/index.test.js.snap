// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Header Component should display the Settings Popup if notifyRemoteConfigWasBlocked is true 1`] = `
<header
  className="root"
>
  <div
    className="content"
  >
    <Link
      className="logo"
      replace={false}
      to="/"
    >
      <h1>
        <span>
          Factom Explorer
        </span>
      </h1>
    </Link>
    <withRouter(Search)
      apiConfig={
        Object {
          "blockchain": "Public",
        }
      }
    />
    <AppMenu />
    <_default
      className="divider"
    />
    <div
      className="blockchainButtonContainer"
    >
      <Button
        className="blockchainButton"
        id="open-settings-popup"
        onClick={[Function]}
        title="Public"
      />
    </div>
  </div>
  <SettingsPopup
    allowCustomCredentials={false}
    apiConfig={
      Object {
        "blockchain": "Public",
      }
    }
    defaultApiConfig={
      Object {
        "blockchain": "Public",
      }
    }
    isConfiguredByDefault={[Function]}
    onClose={[Function]}
    onSubmit={[Function]}
    show={true}
  />
</header>
`;
